platform/viewer/src/appExtensions/MeasurementsPanel/init.js

```diff
@@ -131,15 +40,40 @@ export default function init({ servicesManager, configuration = {} }) {
     this,
     'labelmapModified'
   );
-  //
-  const onRightClick = getOnRightClickCallback(store);
-  const onTouchPress = getOnTouchPressCallback(store);
-  const onNewImage = getResetLabellingAndContextMenu(store);
-  const onMouseClick = getResetLabellingAndContextMenu(store);
-  const onTouchStart = getResetLabellingAndContextMenu(store);
-
-  // Because click gives us the native "mouse up", buttons will always be `0`
-  // Need to fallback to event.which;
+
+  const onRightClick = event => {
+    commandsManager.runCommand('showContextMenu', {
+      event: event.detail,
+    });
+  };
+
+  const onTouchPress = event => {
+    commandsManager.runCommand('showContextMenu', {
+      event: event.detail,
+      props: {
+        isTouchEvent: true,
+      },
+    });
+  };
+
+  const onTouchStart = () => {
+    commandsManager.runCommand('hideContextMenu');
+    commandsManager.runCommand('hideLabellingFlow');
+  };
+
+  const onMouseClick = () => {
+    commandsManager.runCommand('hideContextMenu');
+    commandsManager.runCommand('hideLabellingFlow');
```

Comment:
Commands added by this extension, and not service methods? Not arguing, this just caught me off-guard. I'll have to find their implementation before I have better thoughts.

Reply:
Done! thanks. Now we are using services. :)